services/protocols/MS-Graph-API.js

```diff
@@ -0,0 +1,254 @@
+
+    // Create draft
+    const immutableId = await _createDraftEmail(
+      client,
+      userPrincipalName,
+      mailProperties
+    );
+
+    // Send draft
+    await _sendDraftEmail(client, userPrincipalName, immutableId);
+
+    // Get sent email
+    const response = await _getSentEmail(
+      client,
+      userPrincipalName,
+      immutableId
+    );
+
+    await moveEmailToFolder(MAILBOX_URI, email, "sentbox");
+
+    if (!response.internetMessageId) {
+      console.warn(`No messageId returned for ${email.email} and MS_GRAPH_API`);
+    }
+
+    await updateEmailId(email, response.internetMessageId || "");
+
+    console.log(`Email ${count}: URI = ${email.email}`);
+    console.log(`Email ${count}: Email moved to sentbox`);
+    console.log(`Email ${count}: Email message ID updated`);
+    console.log(
+      `Email ${count}: MessageId updated from ${email.messageId} to ${response.internetMessageId}`
+    );
+  });
+}
+
+/**
+ * Create a draft email using Graph API
+ *
+ * This is necessary to have an email Id we can use for further processing (sending & retrieving)
+ * Note the usage of Prefer IdType="ImmutableId" header, this is so that the id we get for the
+ * draft email does not change as the email gets moved around mailboxes
+ *
+ * API call docs
+ * @see https://docs.microsoft.com/en-us/graph/api/user-post-messages?view=graph-rest-1.0&tabs=javascript
+ *
+ * @param {Client} client Graph API client
+ * @param {string} userPrincipalName The UPN of the user that will be creating the draft email
+ * @param {Object} mailProperties Message object as specified by the Graph API spec @see: https://docs.microsoft.com/en-us/graph/api/resources/message?view=graph-rest-1.0#properties
+ * @returns Immutable id of the created draft email
+ */
+async function _createDraftEmail(client, userPrincipalName, mailProperties) {
+  const response = await client
+    .api(`/users/${userPrincipalName}/messages`)
+    .header("Prefer", 'IdType="ImmutableId"')
+    .post(mailProperties);
+  return response.id;
+}
+
+/**
+ * Send a (draft) email using Graph API
+ *
+ * API call docs
+ * @see https://docs.microsoft.com/en-us/graph/api/message-send?view=graph-rest-1.0&tabs=javascript
+ *
+ * @param {Client} client Graph API client
+ * @param {string} userPrincipalName The UPN of the user we will be sending the email for
+ * @param {string} immutableId Immutable id of the email we will be sending
+ * @returns
+ */
+async function _sendDraftEmail(client, userPrincipalName, immutableId) {
+  return client
+    .api(`/users/${userPrincipalName}/messages/${immutableId}/send`)
+    .post();
+}
+
+/**
+ * Retrieve an email using Graph API
+ *
+ * API call docs
+ * @see https://docs.microsoft.com/en-us/graph/api/message-get?view=graph-rest-1.0&tabs=javascript
+ *
+ * @param {*} client
+ * @param {*} userPrincipalName
+ * @param {*} immutableId
+ * @returns
+ */
+async function _getSentEmail(
+  client,
+  userPrincipalName,
+  immutableId,
+  retry = false
```

Comment:
Fair, the naming isn't ideal. It's indeed meant to allow 1 retry to fetch the email from the sentbox, in case it hasn't appeared yet. Will rename to `mustRetry` and invert the condition